lab-list-comprehensions/your-code/.ipynb_checkpoints/main-checkpoint.ipynb

```diff
@@ -0,0 +1,316 @@
+    "        [0.82579068, 0.91512478, 0.06833034]],\n",
+    "\n",
+    "       [[0.05440634, 0.65857693, 0.30296619],\n",
+    "        [0.06769833, 0.96031863, 0.51293743]],\n",
+    "\n",
+    "       [[0.09143215, 0.71893382, 0.45850679],\n",
+    "        [0.58256464, 0.59005654, 0.56266457]],\n",
+    "\n",
+    "       [[0.71600294, 0.87392666, 0.11434044],\n",
+    "        [0.8694668 , 0.65669313, 0.10708681]],\n",
+    "\n",
+    "       [[0.07529684, 0.46470767, 0.47984544],\n",
+    "        [0.65368638, 0.14901286, 0.23760688]]])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[0.55867166, 0.06210792, 0.08147297, 0.82579068, 0.91512478, 0.06833034, 0.05440634, 0.65857693, 0.30296619, 0.06769833, 0.96031863, 0.51293743, 0.09143215, 0.71893382, 0.45850679, 0.58256464, 0.59005654, 0.56266457, 0.71600294, 0.87392666, 0.11434044, 0.8694668, 0.65669313, 0.10708681, 0.07529684, 0.46470767, 0.47984544, 0.65368638, 0.14901286, 0.23760688]\n"
+     ]
+    }
+   ],
+   "source": [
+    "list5 = [x for y in b for z in y for x in z]\n",
+    "print(list5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 5. Add a condition to the list comprehension above so that the last value in each subarray is printed, but only if it is less than or equal to 0.5."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[0.06210792, 0.08147297, 0.06833034, 0.05440634, 0.30296619, 0.06769833, 0.09143215, 0.45850679, 0.11434044, 0.10708681, 0.07529684, 0.46470767, 0.47984544, 0.14901286, 0.23760688]\n"
+     ]
+    }
+   ],
+   "source": [
+    "lst6= [x for y in b for z in y for x in z if x <= 0.5]\n",
+    "print (lst6)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 6. Use a list comprehension to select and print the names of all CSV files in the */data* directory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "['sample_file_8.txt', 'sample_file_9.txt', 'sample_file_1.csv', 'sample_file_0.csv', 'sample_file_2.csv', 'sample_file_3.csv', 'sample_file_7.csv', 'sample_file_6.csv', 'sample_file_4.csv', 'sample_file_5.csv', 'sample_file_8.csv', 'sample_file_9.csv', 'sample_file_2.txt', 'sample_file_3.txt', 'sample_file_1.txt', 'sample_file_0.txt', 'sample_file_4.txt', 'sample_file_5.txt', 'sample_file_7.txt', 'sample_file_6.txt']\n"
+     ]
+    }
+   ],
+   "source": [
+    "nombres = [nombre for archivos in os.walk('/Users/nayiturbel/Ironhack/1/Labs/Lab-list-comprehension/lab-list-comprehensions/data') for nombre in archivos]\n",
```

Comment:
[nombre for archivos in os.walk('directory') for nombre in archivos if nombre.endswith(csv)]